src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.GCDesc.cs

```diff
@@ -0,0 +1,225 @@
+
+                    // Computing the layout for the boxed version if this is a value type.
+                    int offs = defType.IsValueType ? _type.Context.Target.PointerSize : 0;
+
+                    // Include syncblock
+                    int allocationSize = defType.InstanceByteCount + offs + _type.Context.Target.PointerSize;
+
+                    OutputStandardGCDesc(ref builder, map, allocationSize, offs);
+                    break;
+            }
+
+            Debug.Assert(initialBuilderPosition + GCDescSize == builder.CountBytes);
+        }
+
+        private void OutputStandardGCDesc(ref ObjectDataBuilder builder, GCPointerMap map, int size, int delta)
+        {
+            Debug.Assert(size >= map.Size);
+
+            int pointerSize = _type.Context.Target.PointerSize;
+
+            int numSeries = 0;
+
+            for (int cellIndex = map.Size - 1; cellIndex >= 0; cellIndex--)
+            {
+                if (map[cellIndex])
+                {
+                    numSeries++;
+
+                    int seriesSize = pointerSize;
+
+                    while (cellIndex > 0 && map[cellIndex - 1])
+                    {
+                        seriesSize += pointerSize;
+                        cellIndex--;
+                    }
+
+                    builder.EmitNaturalInt(seriesSize - size);
+                    builder.EmitNaturalInt(cellIndex * pointerSize + delta);
+                }
+            }
+
+            Debug.Assert(numSeries > 0);
+            builder.EmitNaturalInt(numSeries);
+        }
+
+        private void OutputArrayGCDesc(ref ObjectDataBuilder builder, GCPointerMap map)
+        {
+            int numSeries = 0;
+            int leadingNonPointerCount = 0;
+
+            int pointerSize = _type.Context.Target.PointerSize;
+
+            for (int cellIndex = 0; cellIndex < map.Size; cellIndex++)
+            {
+                if (map[cellIndex])
+                    break;
+                leadingNonPointerCount++;
+            }
+
+            int nonPointerCount = leadingNonPointerCount;
+
+            for (int cellIndex = map.Size - 1; cellIndex >= leadingNonPointerCount; cellIndex--)
+            {
+                if (map[cellIndex])
+                {
+                    numSeries++;
+
+                    short pointerCount = 1;
+                    while (cellIndex > 0 && map[--cellIndex])
+                    {
+                        checked { pointerCount++; }
+                    }
+
+                    builder.EmitHalfNaturalInt(pointerCount);
+                    builder.EmitHalfNaturalInt(checked((short)(nonPointerCount * pointerSize)));
+                    nonPointerCount = 0;
+                }
+                else
+                {
+                    nonPointerCount++;
+                }
+            }
+
+            Debug.Assert(numSeries > 0);
+            builder.EmitNaturalInt((2 + leadingNonPointerCount) * pointerSize);
```

Comment:
I still need to figure this out. Any attempts to rationalize the `2 +` part failed so far.

I would expect the `2` to mean `SyncBlock` + `m_pEEType`, but then who ate `_numComponents` that should follow `m_pEEType` and precede the elements?